skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.tooling.projectgen.circuitgen
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import java.io.File
+import java.io.StringWriter
+
+interface CircuitComponent {
+  val fileSuffix: String
+
+  fun generate(packageName: String, className: String): FileSpec
+
+  fun isTestComponent(): Boolean = false
+
+  fun writeToFile(directory: String, packageName: String?, className: String) {
+    val baseDirectory =
+      if (isTestComponent()) directory.replace("src/main", "src/test") else directory
+    val fileSpec = generate(packageName.orEmpty(), className)
+    File(baseDirectory).apply { if (!exists()) mkdirs() }
+    val stringWriter = StringWriter()
+    fileSpec.writeTo(stringWriter)
+    val generatedCode = stringWriter.toString().replace("public ", "")
+    File("${baseDirectory}/${className}${fileSuffix}.kt").writeText(generatedCode)
+  }
+}
+
+class CircuitScreen : CircuitComponent {
+  override val fileSuffix = "Screen"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /*
+    Generate Circuit Screen class, eg:
+
+    @Parcelize
+      class FeatureScreen : Screen {
+        data class State(
+          val message: String = "",
+          val eventSink: Event.() -> Unit = {},
+        ) : CircuitUiState
+
+        @Immutable
+        sealed interface Event : CircuitUiEvent
+      }
+    */
+    val circuitScreenClass = "${className}Screen"
+    val parcelize = ClassName("kotlinx.parcelize", "Parcelize")
+    val circuitUiEvent = ClassName("com.slack.circuit.runtime", "CircuitUiEvent")
+    val circuitUiState = ClassName("com.slack.circuit.runtime", "CircuitUiState")
+    val screenInterface = ClassName("com.slack.circuit.runtime.screen", "Screen")
+    val eventInterface =
+      TypeSpec.interfaceBuilder("Event")
+        .addAnnotation(Immutable::class)
+        .addModifiers(KModifier.SEALED)
+        .addSuperinterface(circuitUiEvent)
+        .build()
+
+    val eventInterfaceName = eventInterface.name?.let { ClassName("", it) }
```

Comment:
Shouldn't the package name be the package name of the outer screen? You can do something like this

```kotlin
val screen = ClassName(<package>, <screen name>)
val eventInterfaceCN = screen.nested("Event")
val eventInterface = TypeSpec.interfaceBuilder(eventInterfaceCN.simpleName)
    // ...
```